types/src/front/assistant/actions/visualization.ts

```diff
@@ -25,3 +25,45 @@ export interface VisualizationActionType extends BaseAction {
 export const VisualizationActionOutputSchema = t.type({
   generation: t.string,
 });
+
+export function visualizationExtractCodeNonStreaming(code: string) {
+  const regex = /<visualization[^>]*>\s*([\s\S]*?)\s*<\/visualization>/;
+  let extractedCode: string | null = null;
+  const match = code.match(regex);
+  if (match && match[1]) {
+    extractedCode = match[1];
+  }
+  if (!extractedCode) {
+    return null;
+  }
+  return extractedCode;
+}
+
+export function visualizationExtractCodeStreaming(code: string) {
+  const startOffset = code.indexOf(">");
```

Comment:
we don't wanna match the whole `<visualization>` instead ?

Reply:
Taken from Aric's PR. I guess we will see when we use it 🙃.